mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMailboxManager.java

```diff
@@ -780,12 +781,23 @@ public void updateAnnotations(MailboxPath mailboxPath, MailboxSession session, L
         for (MailboxAnnotation annotation : mailboxAnnotations) {
             if (annotation.isNil()) {
                 annotationMapper.deleteAnnotation(mailboxId, annotation.getKey());
-            } else {
+            } else if (isValid(mailboxId, annotation, annotationMapper)) {
                 annotationMapper.insertAnnotation(mailboxId, annotation);
             }
         }
     }
 
+    private boolean isValid(MailboxId mailboxId, MailboxAnnotation annotation, AnnotationMapper annotationMapper) throws AnnotationException {
+        if (annotation.size() > MailboxConstants.LIMIT_ANNOTATION_DATA) {
```

Comment:
RFC said "Annotation data size" so it's just value.
It said at: "4.1.  General Considerations"
"The server is allowed to impose limitations on the size of any one
   annotation or the total number of annotations for a single mailbox or
   for the server as a whole.  However, the server MUST accept an
   annotation data size of at least 1024 bytes, and an annotation count
   per server or mailbox of at least 10."